Sources/TSCUtility/SQLite.swift

```diff
@@ -15,6 +15,100 @@ import TSCBasic
 
 /// A minimal SQLite wrapper.
 public struct SQLite {
+    /// The location of the database.
+    public let location: Location
+
+    /// The configuration for the database.
+    public let configuration: Configuration
+
+    /// Pointer to the database.
+    let db: OpaquePointer
+
+    /// Create or open the database at the given path.
+    ///
+    /// The database is opened in serialized mode.
+    public init(location: Location, configuration: Configuration? = nil) throws {
+        self.location = location
+        self.configuration = configuration ?? Configuration()
+
+        var db: OpaquePointer?
+        try Self.checkError("Unable to open database at \(self.location)") {
+            sqlite3_open_v2(
+                location.pathString,
+                &db,
+                SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_FULLMUTEX,
+                nil
+            )
+        }
+
+        self.db = db!
+        try Self.checkError { sqlite3_extended_result_codes(db, 1) }
+        try Self.checkError { sqlite3_busy_timeout(db, self.configuration.busyTimeoutSeconds) }
+    }
+
+    public init(dbPath: AbsolutePath, configuration: Configuration? = nil) throws {
+        try self.init(location: .path(dbPath), configuration: configuration)
+    }
+
+    /// Prepare the given query.
+    public func prepare(query: String) throws -> PreparedStatement {
+        try PreparedStatement(db: self.db, query: query)
+    }
+
+    /// Directly execute the given query.
+    ///
+    /// Note: Use withCString for string arguments.
+    public func exec(query: String, args: [CVarArg] = [], _ callback: SQLiteExecCallback? = nil) throws {
+        let query = withVaList(args) { ptr in
+            sqlite3_vmprintf(query, ptr)
+        }
+
+        let wcb = callback.map { CallbackWrapper($0) }
+        let callbackCtx = wcb.map { Unmanaged.passUnretained($0).toOpaque() }
+
+        var err: UnsafeMutablePointer<Int8>?
+        try Self.checkError { sqlite3_exec(db, query, sqlite_callback, callbackCtx, &err) }
+
+        sqlite3_free(query)
```

Comment:
It took a second to realize that this isn't freeing the same `query` that's passed in as a parameter, but rather a local that shadows it.  Would it be useful to have a different name for the local?

Reply:
Actually I see now that this is existing code that was just moved, so it makes sense to leave as is.

Reply:
addressed